components/server/ee/src/billing/entitlement-service.ts

```diff
@@ -83,20 +84,28 @@ export class EntitlementServiceImpl implements EntitlementService {
     }
 
     async getDefaultWorkspaceTimeout(user: User, date: Date = new Date()): Promise<WorkspaceTimeoutDuration> {
+        let userWorkspaceTimeout: WorkspaceTimeoutDuration | undefined;
```

Comment:
I think all logic around `userWorkspaceTimeout` should not be in here, but in the caller (WorkspaceStarter). This is really only about the "default workspace timeout".

Reply:
Also: When looking how it's used: cant' we use `maySetTimeout` for this? :thinking: 

We only use it [here](https://github.com/gitpod-io/gitpod/pull/16503/files#diff-99c889488d1f0c790ad3f36d2f8d58e94776298c3ad797a278ffc5a36a0979d4R106) to decide whether to use the user preference or the default timeout, no?